client/src/commands/addGatewayIdentityCommand.ts

```diff
@@ -42,52 +45,88 @@ export async function addGatewayIdentity(gatewayItem: GatewayTreeItem): Promise<
-    const openDialogOptions: vscode.OpenDialogOptions = {
-        canSelectFiles: true,
-        canSelectFolders: false,
-        canSelectMany: false,
-        openLabel: 'Select',
-        filters: undefined
-    };
+    let certificate: string;
+    let privateKey: string;
+    let certificatePath: string;
+    let privateKeyPath: string;
 
-    // Get the certificate file path
-    const certPath: string = await UserInputUtil.browseEdit('Browse for a certificate file', quickPickItems, openDialogOptions, gatewayRegistryEntry.name) as string;
-    if (!certPath) {
+    // User selects if they want to add an identity using either a cert/key or an id/secret
+    const addIdentityMethod: string = await UserInputUtil.addIdentityMethod();
+    if (!addIdentityMethod) {
         return Promise.resolve();
     }
-    ParsedCertificate.validPEM(certPath, 'certificate');
-    const keyPath: string = await UserInputUtil.browseEdit('Browse for a private key file', quickPickItems, openDialogOptions, gatewayRegistryEntry.name) as string;
-    if (!keyPath) {
+    const mspID: string = await UserInputUtil.showInputBox('Enter MSP ID');
+    if (!mspID) {
+        // User cancelled entering mspid
         return Promise.resolve();
     }
-    // Get the private key file path
-    ParsedCertificate.validPEM(keyPath, 'private key');
 
-    const FabricWalletGenerator: IFabricWalletGenerator = FabricWalletGeneratorFactory.createFabricWalletGenerator();
-    const wallet: IFabricWallet = FabricWalletGenerator.getNewWallet(gatewayRegistryEntry.name, gatewayRegistryEntry.walletPath);
+    if (addIdentityMethod === UserInputUtil.ADD_CERT_KEY_OPTION) {
+        // User wants to add an identity by providing a certificate and private key
+        const certKey: {certificatePath: string, privateKeyPath: string} = await UserInputUtil.getCertKey(gatewayRegistryEntry.name);
+        if (!certKey) {
+            return Promise.resolve();
+        }
+        certificatePath = certKey.certificatePath;
+        privateKeyPath = certKey.privateKeyPath;
+    } else {
+        // User wants to add an identity by providing a enrollment id and secret
+        const enrollIdSecret: {enrollmentID: string, enrollmentSecret: string} = await UserInputUtil.getEnrollIdSecret();
+        if (!enrollIdSecret) {
+            return Promise.resolve();
+        }
 
-    const certificate: string = await fs.readFile(certPath, 'utf8');
-    const privateKey: string = await fs.readFile(keyPath, 'utf8');
+        const enrollmentID: string = enrollIdSecret.enrollmentID;
+        const enrollmentSecret: string = enrollIdSecret.enrollmentSecret;
 
-    const mspid: string = await UserInputUtil.showInputBox('Enter a mspid');
-    if (!mspid) {
-        // User cancelled entering mspid
+        const enrollment: {certificate: string, privateKey: string} = await FabricCertificateAuthority.enroll(gatewayRegistryEntry.connectionProfilePath, enrollmentID, enrollmentSecret);
```

Comment:
this should be in a try catch. I just gave some BS values for enroll id and secret and nothing happened..